packages/x-date-pickers/src/internals/hooks/usePicker/usePickerViews.ts

```diff
@@ -156,6 +156,23 @@ export const usePickerViews = <
       autoFocus: autoFocusView,
     });
 
+  const handleChange = useEventCallback(
+    (value: TValue, currentViewSelectionState?: PickerSelectionState) => {
+      // If the next view is of different view kind:
+      // `date` views -> `time` views
+      // `time` views -> `date` views
+      const hasNextDifferentKindView =
```

Comment:
For me this assumption is a little too strong
If people want to pass distinct renderers for `year` and `month` instead of passing the same `renderDate` we currently have, then I suspect it would cause issue.

Reply:
Indeed, that could lead to issues if some1 has made VERY custom renderers possibly housing both date and time information in a single layout/view. 🤔 
I'll try to think of something less dangerous. 👌

Reply:
@flaviendelangle What do you think about the updated solution? https://github.com/mui/mui-x/pull/7958/commits/8678f90a3a46dde256e608c78a075308b7ba03b1
IMHO, it should be way more stable. 🙈

Reply:
That's great :ok_hand: